docs/ADR-18-content-mappings-flow-for-explorer-and-builder-in-world.md

```diff
@@ -0,0 +1,262 @@
+
+<!--
+```sequence
+Kernel-&gt;Renderer: Send catalog JSON\nwith all the assets
+Renderer-&gt;Renderer: User places any asset into the world
+Renderer-&gt;Renderer: Map assetId to component id\nin the internal map
+Renderer-&gt;Kernel: Sends special message to\nreflect the assetId mapping
+Renderer-&gt;Kernel: Update component with\n{ src:string } using stateful\nmessages.
+Renderer-&gt;Renderer: Component checks if any assetId\nwas mapped to this component id
+Renderer-&gt;Renderer: Inject mapped ContentProvider to LoadableShape\nuse scene ContentProvider otherwise
+```
+-->
+![resources/adr-18-content-mappings-flow-for-explorer-and-builder-in-world/fig-flow-2-any-user-puts-an-asset-and-the-state-has-to-be-delivered-to-the-stateful-scene.svg](resources/adr-18-content-mappings-flow-for-explorer-and-builder-in-world/fig-flow-2-any-user-puts-an-asset-and-the-state-has-to-be-delivered-to-the-stateful-scene.svg)
+
+Code mock-up of how the integration is going to look like in `LoadableShape` code:
+```csharp
+// Catalog will fill this dictionary. 
+// We can have a specialized catalog class for this responsibility.
+private static Dictionary<string, ContentProvider> assetIds;
+
+// Special message will fill-up this dictionary
+private static Dictionary<string, string> componentToAssetId;
+
+protected virtual void AttachShape(DecentralandEntity entity)
+{
+    ContentProvider provider;
+
+    if (componentToAssetId.ContainsKey(id))
+    {
+        provider = assetIds[componentToAssetId[id]];
+    }
+    else
+    {
+        provider = scene.contentProvider;
+    }
+
+```
+
+This would solve (2) and (3). Trying to solve (1) would involve a re-design of this decision, as we are adding a new message, and this will clash with the scene workers messaging specs.
+
+Here it was discussed about having a monolithic asset catalog message. The monolithic message approach can work in short term, but as new assets are started to be added with the community, this catalog message payload can grow to an unmanageable size. Also, this is already started to happen with our wearables specifications. 
+
+### Migrate all our pipeline to use a more robust asset catalog approach, like the one used by Builder
+
+#### Overview
+
+* Have an asset catalog Kernel<>Renderer messaging pipeline for adding and removing assets lazily as needed.
+* Make all our components receive `assetId` references that are going to be mapped to this catalog to retrieve the proper `ContentProvider` and solve our asset urls.
+* Our `LoadParcelScenes` flow will no longer provide the mappings. All the asset ids management responsibilities are  going to come from the asset catalog flow.
+
+This is arguably the best solution in terms of solving (1) (2) and (3). However, the refactor cost is high. We would have to adapt all the components to this new approach and re-design the LoadParcelScenes message.
+
+We can split this approach in different stages.
+
+#### Stage 1: MVP for builder in-world
```

Comment:
Lets make it clearer that this will only be for the Builder In-World feature. The change to the `LoadParcelScenes` should be backwards compatible, to continue to support scenes deployed right now

Reply:
Added some details to add emphasis on backwards compatibility.